### DISCUSSION TITLE
[vcpkg_buildpath_length_warning] Please add advice to enable long paths on Windows 10

### DISCUSSION THREAD

**Discussion by heydojo** [microsoft/vcpkg#19141] [Ideas]
The text in: https://github.com/microsoft/vcpkg/blob/master/scripts/cmake/vcpkg_buildpath_length_warning.cmake

Could offer a link to the Microsoft document concerning the issue, which is: https://docs.microsoft.com/en-gb/windows/win32/fileio/naming-a-file#enable-long-paths-in-windows-10-version-1607-and-later

This is on the condition that Windows 10 is the host operating system.

Currently, the console just gets spammed with:

```console
CMake Warning at scripts/cmake/vcpkg_buildpath_length_warning.cmake:4 (message):
  qt5-webengine's buildsystem uses very long paths and may fail on your
  system.

  We recommend moving vcpkg to a short path such as 'C:\src\vcpkg' or using
  the subst command.
Call Stack (most recent call first):
  installed/x64-windows/share/qt5/qt_download_submodule.cmake:13 (vcpkg_buildpath_length_warning)
  ports/qt5-webengine/portfile.cmake:72 (qt_download_submodule)
  scripts/ports.cmake:90 (include)
```

Thanks!

**Comment by MVoz** [comment]
jom

Support paths longer than MAX_PATH

https://github.com/microsoft/vcpkg/issues/10200

**Comment by BillyONeal** [comment]
I think we should add such instructions if and only if we have observed that building these ports with that flag enabled in a location wherein the directory structure would result in paths that are too long succeeds with the flag set and fails without.

Note that this probably means build tools and similar would need to be manifested and I'm pretty sure Qt depends on some that are not, but only testing would tell. See https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file#enable-long-paths-in-windows-10-version-1607-and-later for additional details. (I believe there were version(s) of Windows 10 that had bugs where either the flag or the manifest were not correct checked but that was considered a bug)

**Comment by mcraveiro** [comment]
Actually, according to this ticket [1] It appears the MAX_PATH workaround is already enabled in GitHub windows images. Is this correct, and if so, does vcpkg check for the registry change? I am getting this warning on my builds at present.

[1] https://github.com/actions/runner-images/issues/1052

**Reply by Neumann-A** [reply]
The thing is the compiler doesn't support (MSVC) it and if the compiler would support (LLVM) it ninja wouldn't support it..... and after that probably comes a lot of other tools which won't support more than MAX_PATH

**Comment by Neumann-A** [comment]
I think this can be closed since the analyses which was done to include a patched ninja with long paths came to the conclusion that even if ninja would support long paths the compiler tools wouldn't support it. As such enabling long paths on Windows does effectively do nothing unless all the required tools would support long paths on windows.

**Reply by mcraveiro** [reply]
Ah, I see, thanks for the help. Is there a way of suppressing the warning? Given there is nothing one can do as a developer using vcpkg, I rather see either an error when the build or nothing when its just warning me. Thanks for your time.

**Reply by Neumann-A** [reply]
If you use a your own triplet you could define `function(vcpkg_buildpath_length_warning)` and overwrite it but without that there is no other switch.

**Reply by mcraveiro** [reply]
Ah, what a shame. Thanks for the tip though.

**Comment by xiaozhuai** [comment]
https://learn.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation?tabs=registry
Will change `[HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\FileSystem]` do the trick?

**Reply by Neumann-A** [reply]
No
